src/main/java/org/spectrumauctions/sats/core/model/mrvm/MRVMWorld.java

```diff
@@ -55,7 +55,7 @@ public int getNumberOfGoods() {
     }
 
     public Set<MRVMBand> getBands() {
-        return Collections.unmodifiableSet(bands);
+        return ImmutableSet.copyOf(bands);
```

Comment:
Please note that _ImmutableSet.copyOf(collection)_ (which creates a new list) is less performant than _Collections.unmodifiableSet(collection)_ (which creates a view to the exisiting list).  Is there a need to use ImmutableSet?  
The only advantage of using ImmutableCollectionsI can think of in this context would be the clearer contract with the user showing that the returned collection is indeed immutable. To do so, we could change the methods return type to _ImmutableSet_.

Reply:
Note: the cleanest way woud probably by store the class field _bands_ as ImmutableSet, and then always return the same collection (which would be the most performant and cleanest option anyways. But I expect we're not doing this as gson can't handle it and we would have to write too many _TypeAdapters_?

Reply:
Thanks for these hints. I realize that I didn't fully understand the differences between the different ways of making a collection "unmodifiable", mostly throwing in Immutables wherever I could (even though at many points, they don't even make a difference). I'm now trying to find a compromise between unmodifiability, ease of use and performance and will update the PR.